src/main/java/com/onarandombox/multiverseinventories/util/CommentedYamlConfiguration.java

```diff
@@ -66,44 +69,36 @@ public boolean save() {
         if (!doComments) {
             return saved;
         }
+
         // if there's comments to add and it saved fine, we need to add comments
         if (!comments.isEmpty() && saved) {
-            // String array of each line in the config file
-            String[] yamlContents =
-                    this.convertFileToString(file).split("[" + System.getProperty("line.separator") + "]");
-            // This will hold the entire newly formatted config
+            // convert config file to String
+            String stringConfig = this.convertFileToString(file);
+
+            // detect which kind of line endings it uses
+            String lineEnding;
+            if (stringConfig.contains("\r\n")) lineEnding = "\r\n";
+            else lineEnding = "\n";
+
+            // convert stringConfig to array, ignoring the header
+            String[] arrayConfig = stringConfig.substring(stringConfig.indexOf(lineEnding) + lineEnding.length()).split(lineEnding);
+
+            // begin building the new config, starting with the header
             StringBuilder newContents = new StringBuilder();
-            String initialContents = config.options().header();
-            if (initialContents == null) {
-                initialContents = "";
-            }
-            newContents.append(initialContents)
-                    .append(System.getProperty("line.separator"))
-                    .append(System.getProperty("line.separator"));
+            newContents.append("# ").append(config.options().header()).append(lineEnding);
+
```

Comment:
What if there are line breaks in the header?

Reply:
I'm unsure, I'll have to test it out and make sure it's handled correctly. Good catch!

Reply:
No, it won't be handled correctly, I'll have to make a small change.

Reply:
I fixed it, but now I'm experiencing some strange behavior which I can't understand. I changed the line endings to be LF to match [YamlConfiguration](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/browse/src/main/java/org/bukkit/configuration/file/YamlConfiguration.java#119) so that we won't end up making a file with mixed line endings. But if you enter a header with more than one line, the line endings between the lines will be CRLF? I can't figure out where the CR is coming from because it's not YamlConfiguration (the source for the header building method is linked above), and it's not Multiverse-Inventories since I replaced all `System.getProperty("line.separator")` with `\n`... so I'm not sure.

Reply:
Just figured it out. It has to do with feildmaster's Configuration-Library. Gonna see what can be done.

Reply:
Fixed it by removing Configuration-Library. MV Inventories is a little leaner now 😃.